mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -64,6 +64,13 @@ static IntegerAttr mulIntegerAttrs(PatternRewriter &builder, Value res,
   return applyToIntegerAttrs(builder, res, lhs, rhs, std::multiplies<APInt>());
 }
 
+static IntegerOverflowFlagsAttr
+mergeOverflowFlags(IntegerOverflowFlagsAttr val1,
```

Comment:
This could also use a comment of explanation, since the use in canon patterns is not the function declaration.